pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/segment/writer/SegmentWriter.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.ingestion.segment.writer;
+
+import org.apache.pinot.spi.data.readers.GenericRow;
+import org.apache.pinot.spi.env.PinotConfiguration;
+
+
+/**
+ * An interface to write records into Pinot.
+ * This interface helps abstract out details regarding segment generation and push from the caller.
+ */
+public interface SegmentWriter {
```

Comment:
How does this tie in with columnar segment build?

Reply:
Not sure what you mean. That's an implementation specific detail right? For MutableSegmentImpl, we would have a SegmentWriterImpl, such that the `collect` call will call the `realtimeSegment#index` and the `flush` will build the segment exactly as it is being done today.

Reply:
I would rather not do that as part of this PR. There will be several PRs following this, for implementations. We can work on MutableSegmentImpl in one of those